### PR TITLE
Update data-sourcing-project.md

### PR DESCRIPTION
**Pull request by giapippa**
There are some major corrections (typos, grammatical faults) along with some "I would write this way".

### CODE REVIEW

File: data-sourcing-project.md [hunk 1]
```diff
@@ -1,76 +1,74 @@
...
+# Data Dependency Graph in Machine Learning systems
 
 ## Introduction
 
 I was told that the team was always open to project proposals.
 
-This is my first time writting such a proposal, and first time writting an academic proposal, it is
-thus probably riddled with issues, please forgive me for that.
+This is my first time writing such a proposal, and the first time that I write an academic proposal at all.
+I beg your pardon if it is (thus) riddled with issues.
```
Comment (gamazeps) [inline data-sourcing-project.md:8]: Too formal, leave the previous form.

File: data-sourcing-project.md [hunk 2]
```diff
@@ -1,76 +1,74 @@
...
 
 ## Problem statement
 
 ["Hidden Technical Debt in Machine Learning Systems
 D Sculley, Gary Holt, Daniel Golovin, Eugene Davydov, Todd Phillips, Dietmar Ebner, Vinay Chaudhary,
 Michael Young, Jean-Francois Crespo, Dan Dennison", NIPS 2015](https://papers.nips.cc/paper/5656-hidden-technical-debt-in-machine-learning-systems.pdf)
-mentions current technical debt in data analysis systems.
+mentions an existing technical debt in data analysis systems.
```
Comment (gamazeps) [inline data-sourcing-project.md:15]: current is more accurate (and is the term used in the paper).

File: data-sourcing-project.md [hunk 3]
```diff
@@ -1,76 +1,74 @@
...
+  on the correctness of the results.
+- Data Dependencies are usually not easily expressed (to the best of my knowledge): this
   can lead to hidden (or even cyclic) dependencies between data sources.
-- Not knowing the dependencies can lead to broken pipelines when schemas are updated (changing a
-  field in a protobuf for example) which may prove costly to fix.
-  This is also an engineering problem as it may make the importance of each data in the whole system
-  troubling to evaluate (which can be used for promoting the data from spindles to SSDs for example).
+- Ignoring such dependencies can lead to broken pipelines when schemas are updated (e.g. changing a
+  field in a protobuf), which may prove to be costly to fix.
+  Plus, this is also an engineering problem as the importance of data in the whole system
```
Comment (gamazeps) [inline data-sourcing-project.md:28]: Remove the Plus.

File: data-sourcing-project.md [hunk 4]
```diff
@@ -1,76 +1,74 @@
...
+  may appear hard to evaluate (e.g.  promoting the data from spindles to SSDs).
```
Comment (gamazeps) [inline data-sourcing-project.md:29]: Keep the "which can be used", we do not know why we care otherwise.

File: data-sourcing-project.md [hunk 5]
```diff
@@ -1,76 +1,74 @@
...
 - The problem of code dependencies is currently solved using static analysis systems.
 
 ## Objective
 
-An objective of this project would be to have a tool that can create a directed graph of all data
+The objective of this project is to develop a tool that can create a directed graph of all data
```
Comment (gamazeps) [inline data-sourcing-project.md:34]: "An" here is correct, as the reproducibilty of historical pipelines will also be an objective, even though it is not the main one.

File: data-sourcing-project.md [hunk 6]
```diff
@@ -1,76 +1,74 @@
...
 dependencies which could not be bypassed.
 
 ## Proposed approach
 
-A first aproach using the methods from software would be to explicitely write the dependencis in
-code, that could then be analysed with static code analysis tools.
+A first approach using software methods would be to explicitely mention the dependencies in the
```
Comment (gamazeps) [inline data-sourcing-project.md:39]: This change is not correct. We are using the methods *from* software engineering (i.e. static analysis used on software).

File: data-sourcing-project.md [hunk 7]
```diff
@@ -1,76 +1,74 @@
...
+code, allowing to analyse them with static code analysis tools.
 
-This method seems promising but lacks an enforcement policy, and is thus brittle (in the same way
-that CI which accepts code with failing tests do not work from my industry experience).
+This method seems promising but it lacks an enforcement policy, and it is thus brittle (in the same way
+that CI which accept code with failing tests do not work in my industry experience).
```
Comment (gamazeps) [inline data-sourcing-project.md:43]: accepts was correct

File: data-sourcing-project.md [hunk 8]
```diff
@@ -1,76 +1,74 @@
...
 
-A better solution in my opinion would be to enforce this at the executor level using file access
+A better solution, in my opinion, would be to enforce this at the executor level using file access
```
Comment (gamazeps) [inline data-sourcing-project.md:45]: remove first coma

**Review by gamazeps — COMMENTED** [review]
Change according to the comments, otherwise LGTM

File: data-sourcing-project.md (hunk 5 above)
Comment (giapippa) [reply data-sourcing-project.md:34]: I don't like the extreme use of conditional forms ("would", "could", "should", etc). It's like you are not sure at all of what you want to do.

File: data-sourcing-project.md (hunk 7 above)
Comment (giapippa) [reply data-sourcing-project.md:43]: Then it is "does not work" or I did not get the sense of the sentence.

File: data-sourcing-project.md (hunk 6 above)
Comment (giapippa) [reply data-sourcing-project.md:39]: Then explicit it.

**Pull request closed without merging**